projects/plugins/jetpack/_inc/client/main.jsx

```diff
@@ -198,6 +199,75 @@ class Main extends React.Component {
 	}
 
 	renderMainContent = route => {
+		/**
+		 * There are two conditions (groups of conditions, really) where we would want to
+		 * show the partner connection screen.
+		 *
+		 * 1. The site is not yet connected to WPCOM, but has the jetpack_partner_coupon
+		 * option set in the database (this.props.partnerCoupon in redux). This is likely a
+		 * partner-user who has just arrived here from a CTA within a partner's dashboard
+		 * or other ecosystem.
+		 *
+		 * 2. The site is already connected to WPCOM, but the jetpack_partner_coupon option
+		 * is still set in the database. This means the user connected their site, but never
+		 * redeemed the coupon. If this is the case, we don't want to override the dashboard
+		 * or at a glance pages with the redemption screen. Instead, we'll catch a URL
+		 * parameter that JITMs will set (showCouponRedemption=true), and show the screen only
+		 * when the user came from a a JITM.
+		 */
+		if ( this.props.partnerCoupon ) {
+			const forceShow = new URLSearchParams( window.location.search ).get( 'showCouponRedemption' );
+			const partnerCoupon = this.props.partnerCoupon;
+
+			if ( ! this.props.isSiteConnected || forceShow ) {
+				return (
+					<ConnectScreen
+						apiNonce={ this.props.apiNonce }
+						registrationNonce={ this.props.registrationNonce }
+						apiRoot={ this.props.apiRoot }
+						images={ [ '/images/connect-right-partner.png' ] }
+						assetBaseUrl={ this.props.pluginBaseUrl }
+						title={ sprintf(
+							/* translators: %s: Jetpack partner name. */
+							__( 'Welcome to Jetpack %s traveler!', 'jetpack' ),
+							partnerCoupon.partner
+						) }
+						buttonLabel={ sprintf(
+							/* translators: %s: Name of a Jetpack product. */
+							__( 'Set up & redeem %s', 'jetpack' ),
+							partnerCoupon.product.title
+						) }
+						redirectUri={ `admin.php?page=jetpack&partnerCoupon=${ partnerCoupon.coupon_code }` }
+						connectionStatus={ this.props.connectionStatus }
+					>
+						<p>
+							{ sprintf(
+								/* translators: %s: Name of a Jetpack product. */
+								__(
+									'Redeem your coupon and get started with %s for free the first year!',
+									'jetpack'
+								),
+								partnerCoupon.product.title
+							) }
+						</p>
+						<ul>
+							{ partnerCoupon.product.features.map( ( feature, key ) => (
+								<li className="jp-recommendations-product-purchased__feature" key={ key }>
+									{ feature }
+								</li>
+							) ) }
+						</ul>
+						{ this.props.connectionStatus.hasConnectedOwner && (
+							<ActionButton
+								label={ __( 'Redeem coupon', 'jetpack' ) }
```

Comment:
It's funny how you notice details when you're head deep in the code 😄 
The "redeem" name of the button changes depending on the connection status. Should we perhaps change it so both just say "coupon" or both includes the product name?

* `Redeem Jetpack Backup` & `Set up & redeem Jetpack Backup`
* `Redeem coupon` & `Set up & redeem coupon`

Reply:
The design included different copy for each context, and AFAIK this is set up to do what was specified. I'm happy to change it, but might be better to get input from design first.

Reply:
It actually looks like they addressed this in the latest update of the V2 design: p1HpG7-dvB-p2#supporting-use-cases
I'm not 100% sure if it was on purpose or not but I agree with it 😂  We have a decision!